### PR TITLE
network: Record permanent mac address when device is enslaved in a Team, or else /etc/mac-addresses will record broken information

### PR DESCRIPTION
**Pull request by rmetrich**
#### Relax-and-Recover (ReaR) Pull Request Template

Please fill in the following items before submitting a new pull request:

##### Pull Request Details:

* Type: **Bug Fix**

* Impact: **Normal**

* Reference to related issue (URL): See PR https://github.com/rear/rear/pull/1954

* How was this pull request tested? Tested on RHEL7 with a Team on eth0 and eth1

* Brief description of the changes in this pull request:

Use "ethtool -P" as the preferred method to retrieve the MAC address.
Otherwise fall back to other methods, which may lead to some invalid MAC address when using Teams.

### CODE REVIEW

File: usr/share/rear/rescue/GNU/Linux/310_network_devices.sh [hunk 1]
```diff
@@ -854,10 +859,17 @@ function handle_physdev () {
...
     DebugPrint "$network_interface is a physical device"
 
-    if [ -e $sysfspath/bonding_slave/perm_hwaddr ] ; then
-        mac="$( cat $sysfspath/bonding_slave/perm_hwaddr )"
-    else
-        mac="$( cat $sysfspath/address )" || BugError "Could not read a MAC address for '$network_interface'."
+    local mac=""
+
+    if has_binary ethtool ; then
+        mac="$( ethtool -P $network_interface 2>/dev/null | awk '{ print $3 }' )"
```
Comment (pcahyna) [inline usr/share/rear/rescue/GNU/Linux/310_network_devices.sh]: Seems fragile to rely on the MAC address being the third field of ethtool output. What about taking the last field?
```
awk '{print $NF}'
```
(Of course parsing this output which was obviously not designed for anything else than for being seen by a human is horribly barbaric. Why can't ethtool have a more script-friendly output, or why can't sysfs expose this information?)

Comment (rmetrich) [reply usr/share/rear/rescue/GNU/Linux/310_network_devices.sh]: +1

Comment (gozora) [reply usr/share/rear/rescue/GNU/Linux/310_network_devices.sh]: > Why can't ethtool have a more script-friendly output ...

Was asking similar question with `xfs_info` some time ago :-)

V.

**Comment by rmetrich** [comment]
@pcahyna Can you test on your system (I did of course, but just want to be sure)

**Comment by pcahyna** [comment]
@rmetrich I will and report the result. Thanks!

**Comment by pcahyna** [comment]
I tested your patch with and without LACP (teaming only) and it indeed fixed the issue for both cases. I haven't checked bonding for regressions yet.

**Pull request merged**